### DISCUSSION TITLE
Questions around xunit.v3 support for AutoFixture

### DISCUSSION THREAD

**Discussion by evilbaschdi** [xunit/xunit#3006] [Show and tell]
I'm writing .NET Code to earn my money. This also includes writing unit tests.
We also implemented some libraries to make writing tests more ease in our case.

Since these packages are based on **xunit / AutoFixture** (https://github.com/AutoFixture/AutoFixture), I was very interessted in the changes coming with **xunit.v3**.
To be able to evaluate those changes completely "some" would have had implemented xunit.v3 support for **AutoFixture** first.
Because I had some free time on my weekends and during my vacation "some" now is "me".

I also have to say, I neither were involved in developing the xunit nor the AutoFixture projects directly. I only had worked on our libraries so far. So maybe some of my changes could be just wrong.

**So what have I done so far?**
- I've checked out a branch including the latest changes done to **AutoFixture** (https://github.com/AutoFixture/AutoFixture/tree/release/5.0.0)
- I duplicated the project in which some already implemented support for xunit2 and started to work through that "Migrating from v2 to v3" document (https://xunit.net/docs/getting-started/v3/migration)
- I stumbled over that change of namespace / package name thing and - because of that - than renamed the package from "AutoFixture.xUnit3" to "AutoFixture.xUnit.v3"
- At the point "it builds, let's ship it" I pushed my work to my forked repository and created the following Pull request https://github.com/AutoFixture/AutoFixture/pull/1471

As you can see there's already some discussion going on regarding TargetFramework related changes that I did roll back yesterday evening (German time).
Since I'm sure that there are many more topics to discuss, I wanted you as the "xunit community" to maybe help me out 'cause I will not be able to answer them all.

One of that topics is the second thing we're talking about in the PR currently.
As mentioned above I thought I had to change that package name / namespace to "AutoFixture.xUnit.v3" - based on the "Should extension authors follow this same strategy?" paragraph.
But AutoFixture also has some naming conventions I'm currently violating with my latest work.
So does xunit currently require us to change naming strategy completly or does that paragraph only include packages inside that xunit.* namespace?
I understand what your naming strategy is based on but the differentiation is already provided including the version number at the end of their packages.
- AutoFixture.xunit < 2.0.0
- AutoFixture.xunit2 > 2.0.0 < 3.0.0
- AutoFixture.xunit3 > 3.0.0

**Comment by bradwilson** [comment]
> So does xunit currently require us to change naming strategy completly or does that paragraph only include packages inside that xunit.* namespace?

The decision on package naming is completely up to the projects to decide. We gave our reasons for choosing new names so that people could understand why we did it, and they could use that information to help them decide whether they wanted/needed to do that as well. And to be clear, we weren't necessarily suggesting just tagging `.v3` onto the end of a package name, as that's _our_ version number, not _their_ version number.

With that in mind:

> I understand what your naming strategy is based on but the differentiation is already provided including the version number at the end of their packages.
> 
> * AutoFixture.xunit < 2.0.0
> * AutoFixture.xunit2 > 2.0.0 < 3.0.0
> * AutoFixture.xunit3 > 3.0.0

I think that's entirely reasonable.
